src/Components/Components/src/Infrastructure/ComponentStatePersistenceManager.cs

```diff
@@ -48,6 +48,15 @@ public async Task RestoreStateAsync(IPersistentComponentStateStore store)
     /// <param name="renderer">The <see cref="Renderer"/> that components are being rendered.</param>
     /// <returns>A <see cref="Task"/> that will complete when the state has been restored.</returns>
     public Task PersistStateAsync(IPersistentComponentStateStore store, Renderer renderer)
+        => PersistStateAsync(store, renderer.Dispatcher);
+
+    /// <summary>
+    /// Persists the component application state into the given <see cref="IPersistentComponentStateStore"/>.
+    /// </summary>
+    /// <param name="store">The <see cref="IPersistentComponentStateStore"/> to restore the application state from.</param>
+    /// <param name="dispatcher">The <see cref="Dispatcher"/> corresponding to the components' renderer.</param>
+    /// <returns>A <see cref="Task"/> that will complete when the state has been restored.</returns>
+    public Task PersistStateAsync(IPersistentComponentStateStore store, Dispatcher dispatcher)
```

Comment:
I do not think we need this API. We are the only ones that can provide a ComponentStatePersistenceManager to the system. Or am I missing something here?

Reply:
It's a layering thing. The existing caller is in `Microsoft.AspNetCore.Mvc.TagHelpers` and it can't see the internals of `M.A.Components`. It's same reason that you added the other `PersistStateAsync` overload before.